### ISSUE TITLE
Cannot link infix identifiers

### ISSUE THREAD

**Issue by bergus** [haskell/haddock#780]
According to [the docs](http://haskell-haddock.readthedocs.io/en/latest/markup.html#hyperlinked-identifiers), 

> To hyperlink identifiers written in infix form, simply put them in quotes as always: ``'`elem`'``.

This does not work for me.

When I put
```hs
'`Control.Monad.Except.catchError`' (throwError . f)
```
the result is
```html
'<code><a href="../mtl-2.2.1/Control-Monad-Except.html#v:catchError">catchError</a></code>' (throwError . f)
```
Notice that `` ` `` got changed into`'`!

When I swap them and put
```hs
`'Control.Monad.Except.catchError'` (throwError . f)
```
the result is
```html
`<code><a href="../mtl-2.2.1/Control-Monad-Except.html#v:catchError-39-">catchError'</a></code> (throwError . f)
```
which does link the wrong function: `catchError'` instead of `catchError`.

I can kind of make it work by inserting a space, when I put
```hs
`'Control.Monad.Except.catchError' ` (throwError . f)
```
the result is
```html
`<code><a href="../mtl-2.2.1/Control-Monad-Except.html#v:catchError">catchError</a></code> ` (throwError . f)
```
but that looks really ugly.

---

I am on
```
$ stack ghc -- --version
The Glorious Glasgow Haskell Compilation System, version 8.2.2
$ stack --version
Version 1.6.5, Git revision 24ab0d6ff07f28276e082c3ce74dfdeb1a2ca9e9 (5514 commits) x86_64 hpack-0.20.0
```

**Comment by glebec** [comment]
This worked for me:

```hs
-- | @((==) \``on`\` f) x y = f x == f y@
```

_Why_ it worked I am not 100% sure as I don't quite know how to reconcile this with the Haddock documentation.

**Comment by bergus** [comment]
Thanks for this solution @glebec, I had not tried escaping! Unfortunately ``@x \`'on'\` y@`` doesn't work, so I don't understand how it behaves either. (And of course it doesn't solve the original issue of the current behaviour not matching the current documentation).

**Comment by harpocrates** [comment]
``` '`Control.Monad.Except.catchError`' (throwError . f)``` now works as expected:

![screen shot 2019-02-25 at 10 57 27 am](https://user-images.githubusercontent.com/10766081/53361278-3db1a380-38ec-11e9-9749-62d0733fd6d0.png)

Fixed in a5199600c39d25d7b71dcb2328000c1c49ad95a2.
